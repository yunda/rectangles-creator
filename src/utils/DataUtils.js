export default class DataUtils {
    static dbDataToArray(data) {
        const keys = Object.keys(data).filter(key => !!data[key]);
        
        return keys.map(key => ({
            id: key,
            ...data[key]
        }));
    }

    static getTotalWidth(rectangles) {
        return Object.values(rectangles).filter(val => !!val).reduce((sum, rect) => sum + rect.width, 0);
    }
}
