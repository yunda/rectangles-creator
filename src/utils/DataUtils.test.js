import DataUtils from './DataUtils';

test('Should transform db data to array', () => {
    const data = {
        '1': {a: 1},
        '2': {a: 2},
        '3': {a: 3}
    };

    expect(DataUtils.dbDataToArray(data)).toMatchSnapshot();
});

test('Should calculate total width', () => {
    const data = {
        '1': {width: 1},
        '2': {width: 2},
        '3': {width: 3}
    };

    expect(DataUtils.getTotalWidth(data)).toEqual(6);
});
