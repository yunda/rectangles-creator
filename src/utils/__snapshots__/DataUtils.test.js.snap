// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should transform db data to array 1`] = `
Array [
  Object {
    "a": 1,
    "id": "1",
  },
  Object {
    "a": 2,
    "id": "2",
  },
  Object {
    "a": 3,
    "id": "3",
  },
]
`;
